### PR TITLE
Balder/reduce lock time

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst or @toregge PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/multienumattribute.hpp [hunk 1]
```diff
@@ -42,18 +44,31 @@ void
 MultiValueEnumAttribute<B, M>::reEnumerate()
 {
     // update MultiValueMapping with new EnumIndex values.
-    this->logEnumStoreEvent("compactfixup", "drain");
+    this->logEnumStoreEvent("compactfixup", "precompute");
+    vespalib::hash_map<EnumIndex, EnumIndex> old2new(this->_enumStore.getNumUniques());
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/attribute/multienumattribute.hpp]: old2new should not be calculated here.
The mapping from old to new enum indexes should be filled out while performing the compaction in the enum store,
_indexMap should be removed from EnumStoreBase.

**Review by toregge — CHANGES_REQUESTED** [review]

File: searchlib/src/vespa/searchlib/attribute/multienumattribute.hpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/multienumattribute.hpp]: Thanks, done. Please have another look.

File: searchlib/src/vespa/searchlib/attribute/multienumattribute.hpp [hunk 2]
```diff
@@ -41,24 +41,11 @@ MultiValueEnumAttribute<B, M>::considerAttributeChange(const Change & c, UniqueS
 
 template <typename B, typename M>
 void
-MultiValueEnumAttribute<B, M>::reEnumerate()
+MultiValueEnumAttribute<B, M>::reEnumerate(const EnumIndexMap & old2new)
 {
     // update MultiValueMapping with new EnumIndex values.
     this->logEnumStoreEvent("compactfixup", "precompute");
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/attribute/multienumattribute.hpp]: No longer relevant to log precompute

**Review by toregge — APPROVED** [review]
LGTM

**Review by geirst — APPROVED** [review]
👍

**Review by toregge — APPROVED** [review]

**Comment by baldersheim** [comment]
This will be rebased once #7791 is in

**Comment by baldersheim** [comment]
Superceeded by #7795

**Pull request closed without merging**